core_lib/src/interface/timecontrols.cpp

```diff
@@ -209,6 +246,18 @@ void TimeControls::updatePlayState()
     }
 }
 
+void TimeControls::showTimecodeLabelMenu()
+{
+    QPoint globalPos = mTimecodeSelect->cursor().pos();
+
+    QMenu* menu = new QMenu();
+    menu->addAction(tr("No text"), this, &TimeControls::noTimecodeText, 0);
+    menu->addAction(tr("Frames"),  this, &TimeControls::onlyFramesText, 0);
+    menu->addAction(tr("SMPTE Timecode"), this, &TimeControls::SMPTE_text, 0);
+    menu->addAction(tr("SFF Timecode"), this, &TimeControls::SFF_text, 0);
+
+    menu->exec(globalPos);}
```

Comment:
I don't know what you ask for. In fact I looked at the palette implementation, that is listed below.
To me it is (almost) a 1:1 copy. What is it I don't see?
```
void ColorPaletteWidget::showContextMenu(const QPoint& pos)
{
    QPoint globalPos = ui->colorListWidget->mapToGlobal(pos);

    QMenu* menu = new QMenu;
    connect(menu, &QMenu::triggered, menu, &QMenu::deleteLater);

    menu->addAction(tr("Add"), this, &ColorPaletteWidget::addItem, 0);
    menu->addAction(tr("Replace"),  this, &ColorPaletteWidget::replaceItem, 0);
    menu->addAction(tr("Remove"), this, &ColorPaletteWidget::removeItem, 0);

    menu->exec(globalPos);
}
```

Reply:
What I mean is that, instead of connecting a custom showContextMenu slot and manually creating the menu there, you can simply add the actions directly to the QToolButton (mTimecodeSelect) itself. Then you can remove the showContextMenu method entirely and QToolButton will take care of everything that’s necessary to actually show the menu. I will submit suggestions to show what I mean. It probably would have been easier if I had simply done that in the first place, sorry.